go-controller/pkg/ovn/egressgw.go

```diff
@@ -528,3 +563,74 @@ func cleanUpBFDEntry(gatewayIP, gatewayRouter string) {
 		}
 	}
 }
+
+// setExGatewayBridgeForNode registers the availability of a
+// dedicated bridge for external gateway traffic.
+func (oc *Controller) setExGatewayBridgeForNode(node string, secondBridge bool) {
+	oc.nodesExGwInterfaceLock.Lock()
+	defer oc.nodesExGwInterfaceLock.Unlock()
+	oc.nodesExGwInterface[node] = secondBridge
```

Comment:
why the need to cache this? Cant we just do get the node from the lister and check the annotation? Is that too expensive for every pod add?

Reply:
im concerned about adding dependencies between handlers if we dont need to

Reply:
haven't thought about that in first place. I will change using the lister, but I don't believe it won't introduce the same dependency we introduce with a mutex protected map